src/async.rs

```diff
@@ -120,22 +120,11 @@ impl AsyncClient {
 
     /// Get a [`BlockHeader`] given a particular block height.
     pub async fn get_header(&self, block_height: u32) -> Result<BlockHeader, Error> {
-        let resp = self
-            .client
-            .get(&format!("{}/block-height/{}", self.url, block_height))
-            .send()
-            .await?;
-
-        if let StatusCode::NOT_FOUND = resp.status() {
-            return Err(Error::HeaderHeightNotFound(block_height));
-        }
-        let bytes = resp.bytes().await?;
-        let hash =
-            std::str::from_utf8(&bytes).map_err(|_| Error::HeaderHeightNotFound(block_height))?;
+        let block_hash = self.get_block_hash(block_height).await?.to_string();
```

Comment:
nit: I think clippy will complain otherwise since `BlockHash` implements `Display`.
```suggestion
        let block_hash = self.get_block_hash(block_height).await?;
```